Controller/Adminhtml/Menu/Delete.php

```diff
@@ -1,82 +1,45 @@
 <?php
+declare(strict_types=1);
 
 namespace Snowdog\Menu\Controller\Adminhtml\Menu;
 
 use Magento\Backend\App\Action;
-use Magento\Framework\Api\FilterBuilderFactory;
-use Magento\Framework\Api\Search\FilterGroupBuilderFactory;
-use Magento\Framework\Api\SearchCriteriaBuilderFactory;
 use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Controller\ResultInterface;
 use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\Framework\Exception\NotFoundException;
 use Snowdog\Menu\Api\MenuRepositoryInterface;
-use Snowdog\Menu\Api\NodeRepositoryInterface;
 
 class Delete extends Action
 {
+    public const ADMIN_RESOURCE = 'Snowdog_Menu::menus';
+
     /**
      * @var MenuRepositoryInterface
      */
     private $menuRepository;
-    /**
-     * @var NodeRepositoryInterface
-     */
-    private $nodeRepository;
-    /**
-     * @var FilterBuilderFactory
-     */
-    private $filterBuilderFactory;
-    /**
-     * @var FilterGroupBuilderFactory
-     */
-    private $filterGroupBuilderFactory;
-    /**
-     * @var SearchCriteriaBuilderFactory
-     */
-    private $searchCriteriaBuilderFactory;
 
     public function __construct(
         Action\Context $context,
-        MenuRepositoryInterface $menuRepository,
-        NodeRepositoryInterface $nodeRepository,
-        FilterBuilderFactory $filterBuilderFactory,
-        FilterGroupBuilderFactory $filterGroupBuilderFactory,
-        SearchCriteriaBuilderFactory $searchCriteriaBuilderFactory
+        MenuRepositoryInterface $menuRepository
     ) {
         parent::__construct($context);
         $this->menuRepository = $menuRepository;
-        $this->nodeRepository = $nodeRepository;
-        $this->filterBuilderFactory = $filterBuilderFactory;
-        $this->filterGroupBuilderFactory = $filterGroupBuilderFactory;
-        $this->searchCriteriaBuilderFactory = $searchCriteriaBuilderFactory;
     }
 
     /**
      * Dispatch request
      *
-     * @return \Magento\Framework\Controller\ResultInterface|ResponseInterface
-     * @throws \Magento\Framework\Exception\NotFoundException
+     * @return ResultInterface|ResponseInterface
+     * @throws NotFoundException
```

Comment:
Where is this exception (`NotFoundException`) thrown? I see only `CouldNotDeleteException`, `InvalidArgumentException`, `NoSuchEntityException`.